src/WaCore.Sample/Controllers/BooksController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using WaCore.Sample.Data;
+using WaCore.Sample.Entities;
+
+namespace WaCore.Sample.Controllers
+{
+    [Route("api/[controller]")]
+    public class BooksController : Controller
+    {
+        private IUnitOfWork _unitOfWork;
+
+        public BooksController(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        // GET api/books
+        [HttpGet]
+        public async Task<IEnumerable<Book>> Get()
+        {
+            return await _unitOfWork.BooksRepository.GetAllAsync();
+        }
+
+        // GET api/books/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult> GetAsync(int id)
+        {
+            var book = await _unitOfWork.BooksRepository.GetAsync(id);
+            if (book == null)
+                return NotFound();
+            return Json(book);
+        }
+
+        // POST api/books
+        [HttpPost]
+        public async Task<Book> PostAsync([FromBody]Book book)
+        {
+            using (var transaction = await _unitOfWork.BeginTransactionAsync())
+            {
+                _unitOfWork.BooksRepository.Add(book);
```

Comment:
Let's work here with at least 2 repos. Cause currently there is no sense to have a transaction here.